app/scripts/views/sign_in_recovery_code.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AuthErrors = require('../lib/auth-errors');
+const Cocktail = require('cocktail');
+const FormView = require('./form');
+const SignInMixin = require('./mixins/signin-mixin');
+const ServiceMixin = require('./mixins/service-mixin');
+const Template = require('templates/sign_in_recovery_code.mustache');
+const VerificationReasonMixin = require('./mixins/verification-reason-mixin');
+
+const CODE_INPUT_SELECTOR = 'input.recovery-code';
+const VALID_RECOVERY_CODE_REGEX = /^([a-zA-Z0-9]{10})|([a-fA-F0-9]{8})/;
+
+const View = FormView.extend({
+  className: 'sign-in-recovery-code',
+  template: Template,
+
+  beforeRender() {
+    const account = this.getSignedInAccount();
+    if (! account || ! account.get('sessionToken')) {
+      this.navigate(this._getAuthPage());
+    }
+  },
+
+  submit() {
+    const account = this.getSignedInAccount();
+    const code = this.getElementValue('input.recovery-code').toLowerCase();
```

Comment:
Two things:

* The code should be trimmed before being checked and sent to the server.
* You are calling `toLowerCase`, though the validation regexp checks against uppercase letters too.